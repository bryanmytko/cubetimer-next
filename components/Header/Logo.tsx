import Link from "next/link";
import Image from "next/image";
import styles from "./Logo.module.css";

const Logo = () => (
  <Link href="/" className="flex items-center">
    <Image alt="logo" className="h-10 p-1" src="/assets/logo.png" />
    <div className="text-xl">cubetimer.io</div>
  </Link>
);

export default Logo;
