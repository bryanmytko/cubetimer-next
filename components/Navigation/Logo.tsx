import styles from './Logo.module.css';

const Logo = () => (
  <a href="/" className="flex items-center">
    <img className="h-10 p-1" src="/assets/logo.png" />
    <div className="text-xl">cubetimer.io</div>
  </a>
);

export default Logo;