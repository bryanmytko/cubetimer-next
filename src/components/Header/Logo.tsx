import Link from "next/link";
import Image from "next/image";

interface LogoProps {
  dark?: boolean;
}

const Logo = (props: LogoProps) => {
  const { dark } = props;

  return (
    <Link href="/" className="flex items-center mt-1 md:mt-0">
      <Image
        height="47"
        width="45"
        alt="logo"
        className="h-10 p-1"
        src="/assets/logo.png"
      />
      <div className={`text-xl ${dark ? "text-black" : "text-white"}`}>
        cubetimer.io
      </div>
    </Link>
  );
};

export default Logo;
