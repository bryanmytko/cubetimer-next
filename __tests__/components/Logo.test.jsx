import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import "@testing-library/jest-dom";

import Logo from "../../components/Header/Logo";

describe("<Logo>", () => {
  it("renders the component", () => {
    const { container } = render(<Logo />);

    expect(container).toMatchSnapshot();
  });
});
