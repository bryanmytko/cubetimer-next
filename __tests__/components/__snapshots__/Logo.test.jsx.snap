// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo> renders the component 1`] = `
<div>
  <a
    class="flex items-center mt-1 md:mt-0"
    href="/"
  >
    <img
      alt="logo"
      class="h-10 p-1"
      data-nimg="1"
      decoding="async"
      height="47"
      loading="lazy"
      src="/_next/image?url=%2Fassets%2Flogo.png&w=96&q=75"
      srcset="/_next/image?url=%2Fassets%2Flogo.png&w=48&q=75 1x, /_next/image?url=%2Fassets%2Flogo.png&w=96&q=75 2x"
      style="color: transparent;"
      width="45"
    />
    <div
      class="text-xl text-white"
    >
      cubetimer.io
    </div>
  </a>
</div>
`;
